openexr-rs/src/version.rs

```diff
@@ -0,0 +1,309 @@
+use openexr_sys as sys;
+use std::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign};
+use std::os::raw::{c_char, c_int};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct Flag(c_int);
+
+impl BitAnd for Flag {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        Self(self.0.bitand(rhs.0))
+    }
+}
+
+impl BitAndAssign for Flag {
+    fn bitand_assign(&mut self, rhs: Self) {
+        self.0.bitand_assign(rhs.0)
+    }
+}
+
+impl BitOr for Flag {
+    type Output = Self;
+
+    fn bitor(self, rhs: Self) -> Self::Output {
+        Self(self.0.bitor(rhs.0))
+    }
+}
+
+impl BitOrAssign for Flag {
+    fn bitor_assign(&mut self, rhs: Self) {
+        self.0.bitor_assign(rhs.0)
+    }
+}
+
+impl BitXor for Flag {
+    type Output = Self;
+
+    fn bitxor(self, rhs: Self) -> Self::Output {
+        Self(self.0.bitxor(rhs.0))
+    }
+}
+
+impl BitXorAssign for Flag {
+    fn bitxor_assign(&mut self, rhs: Self) {
+        self.0.bitxor_assign(rhs.0)
+    }
+}
+
+impl Flag {
+    /// The flag object is supported by the bound OpenEXR library.
+    pub fn supports_flags(&self) -> bool {
+        let mut result: bool = false;
+
+        unsafe {
+            // NOTE: This will not raise an error, so error checking is skipped.
+            sys::Imf_supportsFlags(&mut result, self.0);
+        }
+
+        result
+    }
+}
+
+/// File is tiled
+pub static TILED_FLAG: Flag = Flag(0x00000200);
+/// File contains long attribute or channel names
+pub static LONG_NAMES_FLAG: Flag = Flag(0x00000400);
+/// File has at least one part which is not a regular scanline image or regular tiled image (that is, it is a deep format)
+pub static NON_IMAGE_FLAG: Flag = Flag(0x00000800);
+/// File has multiple parts
+pub static MULTI_PART_FILE_FLAG: Flag = Flag(0x00001000);
+
+/// Bitwise OR of all known flags.
+pub static ALL_FLAGS: Flag =
+    Flag(TILED_FLAG.0 | LONG_NAMES_FLAG.0 | NON_IMAGE_FLAG.0 | MULTI_PART_FILE_FLAG.0);
+
+/// The version represents a version number and the flags that are enabled for the file. For example,
```

Comment:
Small style thing - I've been adding an empty line underneath the doc comment which makes it more readable imo:
```rust
/// Some comment here
/// # With stuff
///
pub struct Version {}
```